secure_fw/partitions/crypto/crypto_aead.c

```diff
@@ -133,6 +133,13 @@ psa_status_t tfm_crypto_aead_encrypt_setup(psa_invec in_vec[],
     psa_algorithm_t alg = iov->alg;
     mbedtls_svc_key_id_t encoded_key;
 
+    if (handle != TFM_CRYPTO_INVALID_HANDLE)
+    {
+        status = PSA_ERROR_BAD_STATE;
+        *handle_out = handle;
+        goto exit;
+    }
```

Comment:
Releasing the allocated handle in this case is not the right fix I think.
Once the previous setup function has returned success it should either go through a successful finish, or an abort function. 
This is the documented behavior: https://armmbed.github.io/mbed-crypto/html/api/ops/aead.html#c.psa_aead_encrypt_setup
> After a successful call to psa_aead_encrypt_setup(), the application must eventually terminate the operation. The following events terminate an operation:
> 
> A successful call to psa_aead_finish().
> A call to psa_aead_abort().

The abort functionality does more validation of the input handle, so this seems like a much safer approach. Blindly releasing the given handle like this seem unsafe.

```suggestion
    if (handle != TFM_CRYPTO_INVALID_HANDLE)
    {
        *handle_out = handle;
        return PSA_ERROR_BAD_STATE;
    }
```

Reply:
I agree with @joerchan here.
I tested this proposed solution locally successfully.